### PR TITLE
CCP Red Dragon - SSU Fixes + Tests + Smart Gate

### PR DESCRIPTION
**Pull request by CCP-Red-Dragon**
- Fixes for the SSU example and clearer logs for the execute script.
- Changing line endings for Windows users
- Update all readme's
- Added base tests for the smart turret + gate 
- Added tests for the SSU example
- Added a basic smart gate example
- Altered the mprocs to allow for easier testing / development

### CODE REVIEW

File: mock-data/.env [hunk 1]
```diff
@@ -7,5 +7,4 @@ PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
 # Rpc url for local blockchain node
 RPC_URL=http://127.0.0.1:8545
 
-
```
Comment (0xxlegolas) [inline mock-data/.env:10]: can we remove mock-data folder entirely. we might need it anymore

Comment (CCP-Red-Dragon) [reply mock-data/.env:10]: Have removed :)

File: smart-gate/readme.md [hunk 2]
```diff
@@ -3,7 +3,9 @@
 ## Introduction
 This guide will walk you through the process of building contracts for the Smart Gate, deploying them into an existing world running in Docker, and testing their functionality by executing scripts.
```
Comment (0xxlegolas) [inline smart-gate/readme.md]: no that we have the new flow for local deployment we might need changes to readme where they can just run 
`pnpm run dev` instead of having to deploy to the existing world. 
also this section needs update  ### Step 2: Mock data for the existing world **(Local Development Only)**

Comment (0xxlegolas) [reply smart-gate/readme.md]: and add command to run the tests instead of having to mock data

File: smart-storage-unit/packages/contracts/mud.config.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { defineWorld } from "@latticexyz/world";
 
 export default defineWorld({
-  namespace: "test",
+  namespace: "rddragon",
```
Comment (0xxlegolas) [inline smart-storage-unit/packages/contracts/mud.config.ts]: can we change the namespace back to test

File: smart-gate/packages/contracts/mud.config.ts [hunk 4]
```diff
@@ -1,7 +1,14 @@
 import { defineWorld } from "@latticexyz/world";
 
 export default defineWorld({
-  namespace: "test",
+  namespace: "rddragon",
```
Comment (0xxlegolas) [inline smart-gate/packages/contracts/mud.config.ts]: can we revert this change

File: smart-storage-unit/packages/contracts/.env [hunk 5]
```diff
@@ -19,18 +19,17 @@ PLAYER_PRIVATE_KEY=0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b7
 RPC_URL="http://127.0.0.1:8546"
```
Comment (0xxlegolas) [inline smart-storage-unit/packages/contracts/.env]:
```suggestion
RPC_URL="http://127.0.0.1:8545"
```

File: smart-storage-unit/packages/contracts/src/systems/SmartStorageUnitSystem.sol [hunk 6]
```diff
@@ -87,17 +87,14 @@ contract SmartStorageUnitSystem is System {
...
+    uint256 calculatedInput = quantity-quantityInputItemLeftOver;
+
     uint256 itemObjectIdOut = RatioConfig.getItemOut(smartObjectId, inventoryItemIdIn);
 
     TransferItem[] memory inItems = new TransferItem[](1);
-    inItems[0] = TransferItem(inventoryItemIdIn, ssuOwner, quantity);
+    inItems[0] = TransferItem(inventoryItemIdIn, ssuOwner, calculatedInput);
 
     TransferItem[] memory ephTransferItems = new TransferItem[](1);
-    ephTransferItems[0] = TransferItem(itemObjectIdOut, ssuOwner, quantityInputItemLeftOver);
```
Comment (0xxlegolas) [inline smart-storage-unit/packages/contracts/src/systems/SmartStorageUnitSystem.sol:96]: can you explain whats the bug here and how this change fixes the bug ?

File: smart-storage-unit/packages/contracts/.env (hunk 5 above)
Comment (CCP-Red-Dragon) [reply smart-storage-unit/packages/contracts/.env]: Have changed :)

File: smart-storage-unit/packages/contracts/src/systems/SmartStorageUnitSystem.sol (hunk 6 above)
Comment (CCP-Red-Dragon) [reply smart-storage-unit/packages/contracts/src/systems/SmartStorageUnitSystem.sol:96]: It was due to the way that items were being calculated potentially - but doing a different method lets us simplify it to less transfers :)

Comment (0xxlegolas) [reply smart-storage-unit/packages/contracts/src/systems/SmartStorageUnitSystem.sol:96]: inItems is the quantity the user wants to drop in the inventory, I dont understand why that needs to be changes or calculated

**Review by 0xxlegolas — APPROVED** [review]
